functions/query/query.go

```diff
@@ -153,3 +152,72 @@ func toNumber(input string, defaultValue int) (int, error) {
 
 	return strconv.Atoi(input)
 }
+
+func validateRequest(eventBody []byte) (*request, error) {
+
+	requestMap := make(map[string]interface{})
+	if err := json.Unmarshal(eventBody, &requestMap); err != nil {
+		return nil, errors.Wrap(err, "Request body must be a valid JSON object")
+	}
+
+	request := request{}
+	metric, ok := requestMap["metric"]
+	delete(requestMap, "metric")
+	if !ok {
+		return nil, errors.New("Request object is missing 'metric' field")
```

Comment:
why force passing a metric? user can potentially want to query based on filter alone for example for all metric `filter='1==1'`